app/experimenter/static/js/components/BranchManager.js

```diff
@@ -40,25 +40,15 @@ class BranchManager extends React.PureComponent {
     );
   }
 
-  renderBranches() {
-    const { branches } = this.props;
-
-    return branches
-      .withMutations(mutable => {
-        // Make sure the control branch is the first branch
-        mutable
-          .filter(b => b.get("is_control"))
-          .concat(branches.filter(b => !b.get("is_control")));
-      })
-      .map(this.renderBranch);
-  }
-
   render() {
-    const { onAddBranch } = this.props;
+    const { onAddBranch, branches } = this.props;
+    const controlBranch = branches.get(0);
+    const treatmentBranches = branches.slice(1);
 
     return (
       <React.Fragment>
-        {this.renderBranches()}
+        {this.renderBranch(controlBranch, 0)}
+        {treatmentBranches.map((b, i) => this.renderBranch(b, i + 1))}
```

Comment:
probably a dumb question, but is it cause of ordering that we separate the control branch from the rest?

Reply:
Oh you're right I don't even need that! I had an in between state where I was putting control first on the client and the server, but now we don't even need this, good catch.